src/Games/NexusMods.Games.DarkestDungeon/DarkestDungeon.cs

```diff
@@ -45,22 +46,22 @@ public override GamePath GetPrimaryFile(GameStore store)
         );
     }
 
-    protected override IEnumerable<KeyValuePair<GameFolderType, AbsolutePath>> GetLocations(
-        IFileSystem fileSystem,
+    protected override IReadOnlyDictionary<GameFolderType, AbsolutePath> GetLocations(IFileSystem fileSystem,
         IGameLocator locator,
         GameLocatorResult installation)
     {
-        yield return new KeyValuePair<GameFolderType, AbsolutePath>(GameFolderType.Game, installation.Path);
-
-
-        if (installation.Metadata is SteamLocatorResultMetadata { CloudSavesDirectory: not null } steamLocatorResultMetadata)
-            yield return new KeyValuePair<GameFolderType, AbsolutePath>(GameFolderType.Saves, steamLocatorResultMetadata.CloudSavesDirectory.Value);
-
         var globalSettingsFile = fileSystem
             .GetKnownPath(KnownPath.LocalApplicationDataDirectory)
             .Combine("Red Hook Studios/Darkest/persist.options.json");
 
-        yield return new KeyValuePair<GameFolderType, AbsolutePath>(GameFolderType.Preferences, globalSettingsFile);
+        var result =  new Dictionary<GameFolderType, AbsolutePath>()
+        {
+            { GameFolderType.Game, installation.Path },
+            { GameFolderType.Preferences, globalSettingsFile }
+        };
+
+        if (installation.Metadata is SteamLocatorResultMetadata { CloudSavesDirectory: not null } steamLocatorResultMetadata)
+            result[GameFolderType.Saves] = steamLocatorResultMetadata.CloudSavesDirectory.Value;
```

Comment:
There's a `// TODO: Steam Cloud Saves` comment below inside the branch, but this code is right above. Do you know if there a specific known reason why this specific statement is left outside of the branch? 

CC. @erri120 (<= wrote the original code)

I imagine `SteamLocatorResultMetadata` can only be returned when the store is Steam.

Reply:
I changed it to something that makes more sense to me

Reply:
Sure that works.